src/main/java/org/spongepowered/api/command/CommandCompletion.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command;
+
+import net.kyori.adventure.text.Component;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.Optional;
+
+/**
+ * A potential completion of a command argument.
+ */
+public interface CommandCompletion {
```

Comment:
I'd argue this should be called `CommandSuggestion` (or similar with "Suggestion") instead.

Reply:
Why? We use completion in other parts of the API (e.g. ValueCompleter) - the only places we use "suggest" is as a verb to suggest completions.

Reply:
We should be using "suggestion" everywhere, really. We've called them "suggestions" in the API for a long time, so it is what people are used to, especially those dealing with Brigadier and other command libraries now too. Now we have a split of terminology.

Reply:
> We should be using "suggestion" everywhere, really. We've called them "suggestions" in the API for a long time, so it is what people are used to, especially those dealing with Brigadier and other command libraries now too. Now we have a split of terminology. 

@kashike This really should have been said some time ago. It's been like this for some time now and just before we're looking to freeze the API is not really an ideal time to say this.

@gabizou @Zidane What do we want to do - rename all "Completers" to "Suggestors" or whatever, or are we going to stick with what we have now. If we're going to do it, we really should do it now.

Reply:
I've gone back and forth a few times looking at various solutions to this naming scheme, but I think that consistently calling everything `Completions` makes sense.

Colloquially, this operation is known as "tab completion", so any name mentioning completions is likely to be familiar to users. JLine implements a similar operation using a `Completer` that provides `Candidates`. Brigadier chooses to go a bit differently by using pure `Suggestion`/`SuggestionProvider` terminology.

The only thing I think should change if we go with `Completion` terminology is `Command.suggestions` -- it clashes with the completion-focused language the API uses elsewhere.

Reply:
Yeah, I had noticed that too - either way we go we should make that sort of thing consistent and so in that case, it would become `Command.complete`.